capture/lib/ot_communication.py

```diff
@@ -83,4 +189,52 @@ def read_ciphertext(self, len_bytes):
         Returns:
             The received ciphertext.
         """
-        return self.target.simpleserial_read("r", len_bytes, ack=False)
+        if self.simple_serial:
+            return self.target.simpleserial_read("r", len_bytes, ack=False)
+        else:
+            while True:
+                read_line = str(self.ser.readline())
```

Comment:
[@milesdai] (https://github.com/milesdai) how could this done more efficiently? Basically, I want to read the ciphertext back from OT.
- There might be some other data which is not yet read, therefore I am doing this in a loop
- I am receiving `b'\x11RESP_OK:{"ciphertext":[22,104,44,54,253,255,255,255,52,0,0,0,0,0,0,0,0,64,0,0,128,24,0,0,0,0,0,0,0,0,0,0,0,0,66,64,106,108,0,32,160,254,1,16,16,0,0,0,152,32,0,16,198,41,0,32,132,254,1,16,202,192,0,32],"ciphertext_length":4} CRC:3671789451\n'`. Is there a better way of parsing this instead of using split?

Reply:
It looks like this code has been revised now, but I think using the json parsing is correct. uJSON will serialize datatypes as valid json after the `RESP_OK:`.